### PR TITLE
Adds Close button and Animation to Notification component

### PR DESCRIPTION
**Pull request by phuson**
#### What does this PR do?
- Adds close button functionality to Notification component.
- Adds animation to showing/hiding notifications.

#### Any background context you want to provide?
Enhancement per request from issue: https://github.com/grommet/grommet/issues/750

### CODE REVIEW

File: src/js/components/Notification.js [hunk 1]
```diff
@@ -3,16 +3,34 @@
...
+  constructor (props, context) {
+    super(props, context);
+
+    this._onClickClose = this._onClickClose.bind(this);
+    this.state = {
+      isHidden: false
+    };
+  }
+
+  _onClickClose () {
```
Comment (alansouzati) [inline src/js/components/Notification.js]: I believe it is better for the caller to close it. All we have to do is to provide a callback function that the caller will be using to remove the Notification. We do the same with Layer and Tip.

File: src/js/components/Notification.js [hunk 2]
```diff
@@ -75,29 +93,51 @@ export default class Notification extends Component {
       );
     }
 
+    let closeButton;
+    if (this.props.closer) {
+      closeButton = (
+        <Box direction="row" align="start" responsive={false} flex={false}>
+          <Button className={`${CLASS_ROOT}__close-button`}
+            plain={true} onClick={this._onClickClose}
```
Comment (alansouzati) [inline src/js/components/Notification.js]: `onClick={this.props.onClose}`

File: src/js/components/Notification.js [hunk 3]
```diff
@@ -75,29 +93,51 @@ export default class Notification extends Component {
...
+            icon={<CloseIcon className={`${CLASS_ROOT}__close`} />}
+            a11yTitle={
+              Intl.getMessage(this.context.intl, 'Close Notification')
```
Comment (alansouzati) [inline src/js/components/Notification.js]: thanks for adding `a11yTitle` 👏

**Comment by phuson** [comment]
Okay, I have replicated the logic of `closer` and `onClose` props from the Layer component. Thanks!

With that said, I still believe that having a built in closing functionality is easier for users to just be able to pick this up and use it without having to implement their own logic to remove/hide the notifications.

**Comment by alansouzati** [comment]
I agree that `display: none` inside the notification component is easier for callers.

There are bunch of questions out there around this topic:

1 - http://stackoverflow.com/questions/24502898/show-or-hide-element-in-react-js
2 - https://github.com/facebook/react/issues/4373

It seems to me that the general recommendation from the React team is to not render the element at all if you want to remove it. So that's the reason why we are doing Layer, Tip and now Notification like this. It feels that is more aligned with the ReactJS way of doing things.

There are situations where `display: none` would be required. For example with stateful components, see this: https://facebook.github.io/react/docs/multiple-components.html#stateful-children.

I don't think Notification is an example of a stateful component.

**Pull request merged**